src/NzbDrone.Core/DecisionEngine/DownloadDecisionComparer.cs

```diff
@@ -74,6 +83,31 @@ private int ComparePreferredWordScore(DownloadDecision x, DownloadDecision y)
             return CompareBy(x.RemoteEpisode, y.RemoteEpisode, remoteEpisode => remoteEpisode.PreferredWordScore);
         }
 
+        private int CompareIndexerPriority(DownloadDecision x, DownloadDecision y)
+        {
+            return CompareBy(x.RemoteEpisode, y.RemoteEpisode, GetIndexerPriority);
+        }
+
+        private int GetIndexerPriority(RemoteEpisode remoteEpisode)
+        {
+            if (remoteEpisode == null  || remoteEpisode.Release == null || remoteEpisode.Release.IndexerId == 0)
+            {
+                return int.MinValue;
+            }
+
+            try
+            {
+                var indexer = _indexersById[remoteEpisode.Release.IndexerId];
+                var settings = indexer.Settings as IIndexerSettings;
+                return settings == null ? int.MinValue : settings.Priority;
```

Comment:
Do we really want the default value to be `int.MinValue`?, I think a neutral value (`0`) would be better.

Reply:
I think we want this to be `0` as well, but should come from the interface discussed in my other comment.